### PR TITLE
Compat with sklearn-dev

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/dask/dask-ml/issues

### CODE REVIEW

**Comment by TomAugspurger** [comment]
I'm not sure I've gotten things entirely correct.

**Comment by stsievert** [comment]
Looks like the issue mentioned in https://github.com/dask/dask-ml/issues/517#issuecomment-500137410 is still present.

> Dask-ML overwrites `QuantileTransformer._check_inputs`, so fixing this will require accepting `copy` as a keyword argument.

File: dask_ml/preprocessing/data.py [hunk 1]
```diff
@@ -245,7 +245,7 @@ def _check_inputs(self, X, accept_sparse_negative=False):
         # TODO: mix of sparse, dense?
         sample = rng.uniform(size=(5, X.shape[1])).astype(X.dtype)
         super(QuantileTransformer, self)._check_inputs(
-            sample, accept_sparse_negative=accept_sparse_negative
+            sample, accept_sparse_negative=accept_sparse_negative, copy=copy
```
Comment (stsievert) [inline dask_ml/preprocessing/data.py]: Looks like `copy` is only a recognized argument for the development version of Scikit-Learn.

Right now, the CI is throwing errors on every test except `sklearn-dev`. The errors look something like

```
        super(QuantileTransformer, self)._check_inputs(
>           sample, accept_sparse_negative=accept_sparse_negative, copy=copy
        )
E       TypeError: _check_inputs() got an unexpected keyword argument 'copy'
```

**Comment by TomAugspurger** [comment]
Doc build failure is due to https://github.com/dask/dask/issues/4911.

**Pull request merged**